index.html

```diff
@@ -3,134 +3,477 @@
+    });
+
+    d3.select('#region-select').on('change', function () {
+      options.region = d3.event.target.value;
+      options.state = 'all';
+      charts.forEach(function (chart) { chart.update();});
+    });
 
-  // FETCH DATA
+ });
+  
+
+  function Chart(selector, xvariable, xtitle, yvariable, ytitle, xAxisFormat, yAxisFormat, pointClass) {    
+    var chart = this;
+
+    chart.xvariable = xvariable;
+    chart.yvariable = yvariable;
 
-  function setup() {
     // SVG
+    chart.svg = d3.select(selector)
+      .append('svg')
+      .attr('width', width + margin.right + margin.left)
+      .attr('height', height + margin.top + margin.bottom)
+      .append('g')
+      .attr('transform', 'translate(' + margin.left + ',' + margin.top + ')');
+  
+    // SCALES 
+    chart.x = d3.scale.linear()
+      .domain([0, d3.max(data, function (d) { return d[xvariable]; })])
+      .range([0, width])
+      .nice();
 
-    svg = d3.select('#chart')
-      .append('svg');
+    chart.y = d3.scale.linear()
+      .domain(d3.extent(data, function (d) { return d[yvariable]; }))
+      .range([height, 0])
+      .nice();
 
-    // SCALES
+    rscale = d3.scale.sqrt()
+      .domain([0, d3.max(data, function (d) {return d.Pop; })])
+      .range([2,10]);
+     
 
     // AXES
+    var padding=0
+    var axispadding=50
+    var xaxispadding=35
 
-    // EVENT HANDLERS
 
-    d3.select('#year-range').on('change', function () {
-      options.yearRange = d3.event.target.value;
-      update();
-    });
+    // Axes
+    var xAxis = d3.svg.axis()
+      .tickSize(-height,0)
+      .scale(chart.x)
+      .orient("bottom")
+      .ticks(10, xAxisFormat);
+
+    var yAxis = d3.svg.axis()
+      .orient("left")
+      .tickSize(-width)
+      .scale(chart.y)
+      .ticks(10, yAxisFormat);
+    
+    var gx = chart.svg.append("g")
+      .attr("class","y axis")
+      .attr("transform","translate(0,"+ height + ")")
+      .call(xAxis)
+      ;
+
+    var gy = chart.svg.append("g")
+      .attr("class","y axis")
+      .attr("transform","translate("+ padding + ",0)")
+      .call(yAxis)
+      ;
+
+    chart.svg.append("text")
+      .attr("text-anchor", "middle")  // this makes it easy to centre the text as the transform is applied to the anchor
+      .attr("class","axislabel")
+      .attr("transform", "translate("+ (-axispadding) +","+(height/2)+")rotate(-90)")  // text is drawn off the screen top left, move down and out and rotate
+      .text(ytitle);
+
+   chart.svg.append("text")
+      .attr("text-anchor", "middle")  // this makes it easy to centre the text as the transform is applied to the anchor
+      .attr("class","axislabel")
+      .attr("transform", "translate("+ (width/2) +","+(height+(xaxispadding))+")")  // centre below axis
+      .text(xtitle);
+
+    chart.barTooltip = d3.tip()
+       .attr('class', 'tooltip')
+       .offset([-6, 0])
+       .html(function(d) { return d.length + ' counties'; });
+
```

Comment:
So there's some nifty functionality in the d3.tip function to place the tooltip to the left/right/bottom of an element instead of just above it (the default). The direction method does that - using slightly odd nomenclature (n - North / e - east / s - south / w - west). Might help for some of your tooltip placement customization.
```js
 chart.barTooltip = d3.tip()
      .attr('class', 'tooltip')
      .offset([-6, 0])
      .direction('e')
      .html(function(d) { return d.length + ' counties'; });
```
You can also do it with a function (i.e. if condition, then direction(e), else direction(w)).
More on that: https://github.com/Caged/d3-tip/blob/master/docs/positioning-tooltips.md